### ISSUE TITLE
Save/capture message "Notice: Data set contains sampled data"

### ISSUE THREAD

**Issue by efh0888** [skardhamar/rga#79]
Is it possible to somehow capture this message, i.e. store it as a character string for later use? I've got a script that automates our data extracts from Google Analytics using the rga package, and a table that keeps logs for each query, so it'd be nice to include a column there indicating whether or not sampling was used.

**Comment by mattpolicastro** [comment]
The Reporting API response does include [fields regarding sampling](https://developers.google.com/analytics/devguides/reporting/core/v3/reference#sampling) (`containsSampledData`, `sampleSize`, and `sampleSpace`) so it's not unfeasible. How would you see this data being returned, though? As another column on the results table?

**Comment by MarkEdmondson1234** [comment]
I put this data as an attribute of the data frame results, so you can reach it via say attr("sampled", garesults)

**Comment by efh0888** [comment]
I think an attribute makes sense

**Comment by mattpolicastro** [comment]
Had no idea attributes were a thing. Seems great.

**Comment by mattpolicastro** [comment]
[Took a first stab at this](https://github.com/mattpolicastro/rga/commit/ecd6fcf8dada7bc1a2a5bb0b04654bd10ac8a980) with a simple conditional block in `core.R`. Thoughts?

**Comment by efh0888** [comment]
Looks good to me, except the additional warning is redundant. Might also want to add something like
```
else {
  attr(ga.data.df, "ga:containsSampledData") <- FALSE
}
```

**Comment by mattpolicastro** [comment]
Yep, the warning was an oversight on my part. [Latest commit.](https://github.com/mattpolicastro/rga/commit/18d405ae1548f8f929e243cbbd2bcecef6f23b86)

**Comment by MarkEdmondson1234** [comment]
I would avoid the ':' as it may do funky things to R names.

As a suggested enhancement, I got this from RGoogleAnalytics and it may be good here - it replicates the "This data is drawn from 2% of visits" messages you get in the GA interface:

```
samplePercent <-  100
  if(!is.null(x$containsSampledData)){
    if(x$containsSampledData) {
      samplePercent <- round(100 * (as.numeric(x$sampleSize) / as.numeric(x$sampleSpace)), 2)
      message("Data is sampled, based on ", samplePercent, "% of visits. Use WALK to mitigate it." )
    }
  }
```

**Comment by mattpolicastro** [comment]
I like it. I'll add something in.

**Comment by EricGoldsmith** [comment]
FWIW, I did something similar in my local fork (and, apologies, never submitted a PR):
https://github.com/tedconf/rga/commit/74dd9a44c5d31d7c99657d3f7a4f15136ad7741e

**Comment by mattpolicastro** [comment]
@EricGoldsmith yeah, that's about what I had in mind. Mind if I pull it into my fork so I can PR it?

**Comment by EricGoldsmith** [comment]
@mattpolicastro, be my guest.

**Comment by mattpolicastro** [comment]
Super. I'll restrict myself to that particular commit as your other changes won't merge as cleanly, but I'll start in on it.

**Comment by mattpolicastro** [comment]
Ok, I think I've incorporated everything [into one branch](https://github.com/skardhamar/rga/compare/master...mattpolicastro:sampling-attributes). Any more thoughts before I submit a PR?

**Comment by MarkEdmondson1234** [comment]
:+1:

**Comment by EricGoldsmith** [comment]
@mattpolicastro, can you confirm that this is now fixed, and then close this issue?

**Comment by mattpolicastro** [comment]
Sorry for not closing the loop on this—it appears to be working correctly after force-updating the package install. However, it looks like @efh0888 may have to close it as I'm not an admin.
